Update G000-G001.md

PostWarStew opened the pull request:
Adding example of single axis homing with S1 option. I had issues with re homing machine between layers with G28. Was able to find older examples using S1 option within G1 (duet3d.dozuki.com/Wiki/ConfiguringRepRapFirmwareCartesianPrinter#Section_Homing_X_and_Y). But did not find any examples of using S1 in your documentation or examples of the canned cycles G28 references or their location within firmware.

thinkyhead commented:
In Marlin the `S` parameter is not used this way because Marlin doesn't have "homing mode" for axes. All homing in Marlin is done through `G28`, and probes use `G29` and `G30`.

In Marlin, the `S` parameter in `G1` is used for setting laser power when the `LASER_FEATURE` is enabled.

PostWarStew commented:
You should double check that.

I had problems with G28 for X and Y only. Z axis did not return to original position (started printing 8mm above previous print level).

I was able to get around this by using the previously stated code to home the axis. As the stated code is what is reference in the G28 canned cycle.

The S option is G code dependent so it maybe interpreted differently depending on the G code that precedes it.

You can check it yourself with any of the printers running Marlin.

G-Code has been my jam since I have been CNC programming for the better part of a decade now.

Regards,
Carlon LaMont

From: Scott ***@***.***>
Sent: Friday, August 5, 2022 10:47 PM
To: ***@***.***>
Cc: ***@***.***>; ***@***.***>
Subject: Re: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR #462)


In Marlin the S parameter is not used this way because Marlin doesn't have "homing mode" for axes. All homing in Marlin is done through G28, and probes use G29 and G30.

In Marlin, the S parameter in G1 is used for setting laser power when the LASER_FEATURE is enabled.

—
Reply to this email directly, view it on GitHub<https://github.com/MarlinFirmware/MarlinDocumentation/pull/462#issuecomment-1207153563>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADK3G2Y75TMBNJNVJQK6THLVXX4ATANCNFSM55L3LWTA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

PostWarStew commented:
Also, this did work for me on Marlin 1.1.9.

I can test it on Marlin 2.0. Was able to update firmware recently.

Currently working on getting Marlin 2.1 to even compile.
Had to make changes to signature.py (Marlin-2.1/buildroot/share/PlatformIO/scripts).
Whoever is making the configure and configure_adv (and the coresponding machine templates) is adding a lot of non utf8 characters. Probably in the comment sections from what I can tell.

            original: f = open(filepath, encoding="utf8").read().split("\n")
            current: f = open(filepath, encoding="utf8", errors='ignore').read().split("\n")

Getting a lot of errors because of:

Marlin\src\module/temperature.h:309:51: error: 'TEMP_HYSTERESIS' was not declared in this scope
Marlin\src\module/temperature.h:309:85: error: template argument 2 is invalid
Marlin\src\module/temperature.h:312:55: error: 'TEMP_BED_HYSTERESIS' was not declared in this scope
Marlin\src\module/temperature.h:312:97: error: template argument 2 is invalid
Marlin\src\module/temperature.h:444:140: error: 'TEMP_WINDOW' was not declared in this scope

That I am still working through.

Regards,
Carlon LaMont

________________________________
From: carlon lamont ***@***.***>
Sent: Saturday, August 6, 2022 11:04 AM
To: MarlinFirmware/MarlinDocumentation ***@***.***>
Subject: RE: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR #462)


You should double check that.



I had problems with G28 for X and Y only. Z axis did not return to original position (started printing 8mm above previous print level).



I was able to get around this by using the previously stated code to home the axis. As the stated code is what is reference in the G28 canned cycle.



The S option is G code dependent so it maybe interpreted differently depending on the G code that precedes it.



You can check it yourself with any of the printers running Marlin.



G-Code has been my jam since I have been CNC programming for the better part of a decade now.



Regards,

Carlon LaMont



From: Scott ***@***.***>
Sent: Friday, August 5, 2022 10:47 PM
To: ***@***.***>
Cc: ***@***.***>; ***@***.***>
Subject: Re: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR #462)



In Marlin the S parameter is not used this way because Marlin doesn't have "homing mode" for axes. All homing in Marlin is done through G28, and probes use G29 and G30.

In Marlin, the S parameter in G1 is used for setting laser power when the LASER_FEATURE is enabled.

—
Reply to this email directly, view it on GitHub<https://github.com/MarlinFirmware/MarlinDocumentation/pull/462#issuecomment-1207153563>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADK3G2Y75TMBNJNVJQK6THLVXX4ATANCNFSM55L3LWTA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

EvilGremlin commented:
You are wrong and trying to argue with main project mainteainer, who have better comprehension of this code than anyone else. Gcode is barely a standard, each firmware have different flavour. RRF or any other docs are absolutely irrelevant to Malrin.

PostWarStew commented:
Spicy!

I apologize and concede. Was not trying to come off argumentative.
Removed S from the G01 command homing example and it still worked.

Great code and thank you for all your work.

I know G-Code is not uniformly standardized. As I have to study
machine-specific G-Code and macros.

Hopefully, I will be contributing in a more meaningful way in the coming
weeks.

Let me know if there is a board or discussion about issues that you could
use help with.

Somehow is signed into another Git account that I don't really use.
https://github.com/MasterofNull

Regards,
Carlon LaMont

P.S. EvilGremlin is a nice touch. Though I can understand it can be pretty
brutal out there, communicating with some folks.

---------- Forwarded message ---------
From: carlon lamont ***@***.***>
Date: Sat, Aug 6, 2022 at 1:05 PM
Subject: Fw: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR
#462)
To: ***@***.*** ***@***.***>


Re: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR #462)
EvilGremlin ***@***.***>
**
**
**
**
To:

   - MarlinFirmware/MarlinDocumentation

Cc:

   - You;
   - Author

Sat 8/6/2022 12:31 PM

You are wrong and trying to argue with main project mainteainer, who have
better comprehension of this code than anyone else. Gcode is barely a
standard, each firmware have different flavour. RRF or any other docs are
absolutely irrelevant to Malrin.

—
Reply to this email directly, view it on GitHub
<https://github.com/MarlinFirmware/MarlinDocumentation/pull/462#issuecomment-1207260854>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ADK3G23FDOTECZIVM4KWHKDVX2VQHANCNFSM55L3LWTA>
.
You are receiving this because you authored the thread


*From:* carlon lamont ***@***.***>

*Sent:* Saturday, August 6, 2022 11:43 AM

*To:* MarlinFirmware/MarlinDocumentation <
***@***.***>
*Subject:* Re: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR
#462)

Also, this did work for me on Marlin 1.1.9.

I can test it on Marlin 2.0. Was able to update firmware recently.

Currently working on getting Marlin 2.1 to even compile.
Had to make changes to signature.py
(Marlin-2.1/buildroot/share/PlatformIO/scripts).
Whoever is making the configure and configure_adv (and the coresponding
machine templates) is adding a lot of non utf8 characters. Probably in the
comment sections from what I can tell.

original: f = open(filepath, encoding="utf8").read().split("\n")
current: f = open(filepath, encoding="utf8",
errors='ignore').read().split("\n")

Getting a lot of errors because of:

Marlin\src\module/temperature.h:309:51: error: 'TEMP_HYSTERESIS' was not
declared in this scope
Marlin\src\module/temperature.h:309:85: error: template argument 2 is
invalid
Marlin\src\module/temperature.h:312:55: error: 'TEMP_BED_HYSTERESIS' was
not declared in this scope
Marlin\src\module/temperature.h:312:97: error: template argument 2 is
invalid
Marlin\src\module/temperature.h:444:140: error: 'TEMP_WINDOW' was not
declared in this scope

That I am still working through.

Regards,
Carlon LaMont

------------------------------
*From:* carlon lamont ***@***.***>
*Sent:* Saturday, August 6, 2022 11:04 AM
*To:* MarlinFirmware/MarlinDocumentation <
***@***.***>
*Subject:* RE: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR
#462)


You should double check that.



I had problems with G28 for X and Y only. Z axis did not return to original
position (started printing 8mm above previous print level).



I was able to get around this by using the previously stated code to home
the axis. As the stated code is what is reference in the G28 canned cycle.



The S option is G code dependent so it maybe interpreted differently
depending on the G code that precedes it.



You can check it yourself with any of the printers running Marlin.



G-Code has been my jam since I have been CNC programming for the better
part of a decade now.



Regards,

Carlon LaMont



*From: *Scott Lahteine ***@***.***>
*Sent: *Friday, August 5, 2022 10:47 PM
*To: *MarlinFirmware/MarlinDocumentation
***@***.***>
*Cc: *PostWarStew ***@***.***>; Author ***@***.***>
*Subject: *Re: [MarlinFirmware/MarlinDocumentation] Update G000-G001.md (PR
#462)



In Marlin the S parameter is not used this way because Marlin doesn't have
"homing mode" for axes. All homing in Marlin is done through G28, and
probes use G29 and G30.

In Marlin, the S parameter in G1 is used for setting laser power when the
LASER_FEATURE is enabled.

—
Reply to this email directly, view it on GitHub
<https://github.com/MarlinFirmware/MarlinDocumentation/pull/462#issuecomment-1207153563>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ADK3G2Y75TMBNJNVJQK6THLVXX4ATANCNFSM55L3LWTA>
.
You are receiving this because you authored the thread.
